src/arb/arb.jl

```diff
@@ -1814,6 +1814,36 @@ function lindep(A::Array{arb, 1}, bits::Int)
   return [L[1, i] for i = 1:n]
 end
 
+################################################################################
+#
+#  Simplest rational inside
+#
+################################################################################
+
+@doc Markdown.doc"""
+      simplest_between(x::arb)
+ 
+> Return the simplest fraction inside the ball `x`. A canonical fraction
+> `a_1/b_1` is defined to be simpler than `a_2/b_2` iff `b_1 < b_2` or > `b_1 =
+> b_2` and `a_1 < a_2`.
+"""
+function simplest_rational_inside(x::arb)
+   a = fmpz()
+   b = fmpz()
+   e = fmpz()
+
+   ccall((:arb_get_interval_fmpz_2exp, libarb), Nothing,
+         (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}, Ref{arb}), a, b, e, x)
+   @assert fits(Int, e)
```

Comment:
An assert is not quite right here. It probably better to throw in this exceptional case because the answer doesn't fit into an fmpq.